Library/Homebrew/test/cask/reinstall_spec.rb

```diff
@@ -13,6 +13,7 @@
       ==> Downloading file:.*caffeine.zip
       Already downloaded: .*--caffeine.zip
       ==> Uninstalling Cask local-caffeine
+      ==> Skipping processing of login_items
```

Comment:
I have a concern that there may be side effects of skipping the uninstallation of `login_items`.
I understand having the option to automatically register a `login_item`, but is the idea here to be able to skip uninstalling them? The reason we generally aim to remove as much as possible with `brew uninstall` (which also runs with `brew upgrade`) is to not leave things behind that aren't connected to the application anymore.

What happens if the `login_item` changes between versions?

Reply:
Agree - I was also facing the same conundrum. This is predominantly for the usecase where the user has chosen to not uninstall by using `--no-login-items` CLI switch. By default, the registered login items will be uninstalled (I think that's how I have coded it?)